netconf.c

```diff
@@ -1169,32 +1374,37 @@ static bool
 handle_unlock (struct netconf_session *session, xmlNode * rpc)
 {
     xmlNode *action = xmlFirstElementChild (rpc);
-    xmlNode *error_info, *node;
-    char *msg = NULL;
-    char session_id_str[32];
+    xmlNode *node;
+    nc_error_parms error_parms = NC_ERROR_PARMS_INIT;
+    GString *error_msg = g_string_new (NULL);
```

Comment:
Does this structure need to be freed at some point? Not all return paths do this I think.